singularity-eos/eos/eos_stiff.hpp

```diff
@@ -0,0 +1,200 @@
+    return _Cv;
+  }
+  PORTABLE_INLINE_FUNCTION Real SpecificHeatFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const {
+    return _Cv;
+  }
+  PORTABLE_INLINE_FUNCTION Real BulkModulusFromDensityTemperature(
+      const Real rho, const Real temperature, Real *lambda = nullptr) const {
+    const Real entr = EntropyFromDensityTemperature(rho, temperature, lambda);
+    return MYMAX(0.0, pow(rho ,_gm1    ) * 
+                      pow(_Cv ,_gm1+1.0) * 
+                      pow(_gm1,_gm1+2.0) * exp((entr-_qp)/_Cv) );
+  }
+  PORTABLE_INLINE_FUNCTION Real BulkModulusFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const {
+    const Real entr = EntropyFromDensityInternalEnergy(rho, sie, lambda);
+    return MYMAX(0.0, pow(rho ,_gm1    ) * 
+                      pow(_Cv ,_gm1+1.0) * 
+                      pow(_gm1,_gm1+2.0) * exp((entr-_qp)/_Cv) );
+  }
+  PORTABLE_INLINE_FUNCTION Real GruneisenParamFromDensityTemperature(
+      const Real rho, const Real temperature, Real *lambda = nullptr) const {
+    return _gm1;
+  }
+  PORTABLE_INLINE_FUNCTION Real GruneisenParamFromDensityInternalEnergy(
+      const Real rho, const Real sie, Real *lambda = nullptr) const {
+    return _gm1;
+  }
+  PORTABLE_INLINE_FUNCTION void FillEos(Real &rho, Real &temp, Real &energy, Real &press,
+                                        Real &cv, Real &bmod, const unsigned long output,
+                                        Real *lambda = nullptr) const;
+  PORTABLE_INLINE_FUNCTION
+  void ValuesAtReferenceState(Real &rho, Real &temp, Real &sie, Real &press, Real &cv,
+                              Real &bmod, Real &dpde, Real &dvdt,
+                              Real *lambda = nullptr) const {
+    // use STP: 1 atmosphere, room temperature
+    rho = _rho0;
+    temp = _T0;
+    sie = _sie0;
+    press = _P0;
+    cv = _Cv;
+    bmod = _bmod0;
+    dpde = _dpde0;
+  }
+  // Generic functions provided by the base class. These contain e.g. the vector
+  // overloads that use the scalar versions declared here
+  SG_ADD_BASE_CLASS_USINGS(StiffGas)
+  PORTABLE_INLINE_FUNCTION
+  int nlambda() const noexcept { return 0; }
+  static constexpr unsigned long PreferredInput() { return _preferred_input; }
+  static inline unsigned long scratch_size(std::string method, unsigned int nelements) {
+    return 0;
+  }
+  static inline unsigned long max_scratch_size(unsigned int nelements) { return 0; }
+  PORTABLE_INLINE_FUNCTION void PrintParams() const {
+    printf("Stiff Gas Parameters:\nGamma = %g\nCv    = %g\nPinf  = %g\nq     = %g\nqprime= %g\n", _gm1 + 1.0, _Cv,_Pinf,_qq,_qp);
+  }
+  PORTABLE_INLINE_FUNCTION void
+  DensityEnergyFromPressureTemperature(const Real press, const Real temp, Real *lambda,
+                                       Real &rho, Real &sie) const {
+    sie = MYMAX(0.0, (press + (_gm1+1.0)*_Pinf)/(press + _Pinf)*_Cv*temp+_qq );
```

Comment:
We'll probably want to just force `press` to be larger than `_Pinf` here. Again, we don't want to floor the energy at zero either.

Reply:
Lower bound adjusted